models/userModel.js

```diff
@@ -37,6 +39,34 @@ class User {
     };
   }
 
+  generateToken(user) {
+    return jwt.sign(
+      {
+        id: user._id,
+        email: user.email,
+        role: user.role,
+      },
+      process.env.SECRET,
+      { expiresIn: "1h" }
+    );
+  }
+
+  async login(username, password) {
+    const user = await this.userModel.findOne({ email: username });
+    if (!user) throw new NotFoundError("Username or password is incorrect");
+
+    const validPassword = bcrypt.compare(password, user.password);
+    if (!validPassword)
+      throw new UnauthorizedError("Username or password is incorrect");
+
+    delete user.password;
+
+    return {
```

Comment:
Should return:
`{
  token: "TJOSSAN",
  user: {
    email: "kdajsdaskj",
    name: "fkdjsofksj",
    role: "customer",
    adress: {
      street: "omg",
      zip: "shiet",
      city: "Smygehuk"
    }
  }
}`